pkg/controller/cluster/upgrades.go

```diff
@@ -171,6 +172,8 @@ func (c *Controller) upgrade424(ctx context.Context, tenant *miniov2.Tenant) (*m
 func versionCompare(version1 string, version2 string) int {
 	version1 = version1[1:]
 	version2 = version2[1:]
+	version1 = strings.Split(version1, "-")[0]
+	version2 = strings.Split(version2, "-")[0]
```

Comment:
This function is quite unsafe TBH it has no protection for input parameters and it is prone to crashes. Looks like some helpful libraries might help here.

Reply:
Is it worth at this point replacing the majority of this logic with a semver library?

Reply:
Yes I think that is a better idea @Rushmead

Reply:
I've changed this to use blang/semver, which is the semver library that is used elsewhere in the minio codebase. As I mention in the commit message, I'm much happier now that this is more semver compliant even though in some cases it would not 100% fix the issue.